### PR TITLE
Support for tag highlighting

### PR DESCRIPTION
**Pull request by me-matt**
Added tag highlighting mirroring the wikilinks mechanism.

### CODE REVIEW

File: packages/foam-vscode/src/features/document-decorator.ts [hunk 1]
```diff
@@ -14,34 +14,59 @@ const placeholderDecoration = vscode.window.createTextEditorDecorationType({
...
+        link.range.start.line,
+        link.range.start.character + (link.type === 'wikilink' ? 2 : 0),
+        link.range.end.line,
+        link.range.end.character - (link.type === 'wikilink' ? 2 : 0)
+      )
+    ) as vscode.Range[];
+
+const getTagRanges = (tags: Tag[]): vscode.Range[] =>
+  tags.map(tag =>
+    Range.create(
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/features/document-decorator.ts:62]: you should be able to use the `toVsCodeRange` function here

File: packages/foam-vscode/src/features/document-decorator.ts [hunk 2]
```diff
@@ -14,34 +14,59 @@ const placeholderDecoration = vscode.window.createTextEditorDecorationType({
   cursor: 'pointer',
 });
 
+const tagDecoration = vscode.window.createTextEditorDecorationType({
+  rangeBehavior: vscode.DecorationRangeBehavior.ClosedClosed,
+  textDecoration: 'none',
+  color: { id: 'foam.tag' },
+  cursor: 'pointer',
```
Comment (riccardoferretti) [inline packages/foam-vscode/src/features/document-decorator.ts:21]: how come a pointer cursor?

**Review by riccardoferretti — COMMENTED** [review]
Looks good, thanks for taking care of this. I left a couple of minor comments, let me know your thoughts

File: packages/foam-vscode/src/features/document-decorator.ts (hunk 2 above)
Comment (me-matt) [reply packages/foam-vscode/src/features/document-decorator.ts]: It wasn't really intentional, just an artifact from the links object. I've removed.

File: packages/foam-vscode/src/features/document-decorator.ts (hunk 1 above)
Comment (me-matt) [reply packages/foam-vscode/src/features/document-decorator.ts]: Good call 👍 Done

File: packages/foam-vscode/src/features/document-decorator.ts [hunk 3]
```diff
@@ -14,33 +14,58 @@ const placeholderDecoration = vscode.window.createTextEditorDecorationType({
   cursor: 'pointer',
 });
 
+const tagDecoration = vscode.window.createTextEditorDecorationType({
+  rangeBehavior: vscode.DecorationRangeBehavior.ClosedClosed,
```
[the remark this answers is not in the document]
Comment (me-matt) [reply packages/foam-vscode/src/features/document-decorator.ts:18]: Sorry for the delay, moved overseas 😂

What are the expectations for the editing behaviour?

[the remark this answers is not in the document]
Comment (riccardoferretti) [reply packages/foam-vscode/src/features/document-decorator.ts:18]: Congrats! I hope you are enjoying your new country!

What I mean here is that if you write `#ta` and you add a `g`, you will get `#tag` which still needs to be highlighted. Basically that's what this setting is for

[the remark this answers is not in the document]
Comment (riccardoferretti) [reply packages/foam-vscode/src/features/document-decorator.ts:18]: Forgot to answer the second part of the question. What I meant by editing behavior was just to do a few testing of adding/removing/expanding/reducing tags and how the highlight responds

**Comment by memeplex** [comment]
Perhaps too late but wouldn't it be better to use a syntax injection? It's simpler, faster and more configurable. AFAICS there's no smart behavior here that requires decorations.

**Comment by riccardoferretti** [comment]
Fair point @memeplex - what do you think @me-matt ?

**Comment by riccardoferretti** [comment]
closing for lack of activity

**Pull request closed without merging**